app/models/template_version.rb

```diff
@@ -43,6 +45,76 @@ def form_json_requirements
     json_requirements
   end
 
+  def publish_event_notification_data
+    {
+      "id" => SecureRandom.uuid,
+      "action_type" => Constants::NotificationActionTypes::NEW_TEMPLATE_VERSION_PUBLISH,
+      "action_text" => "#{label} - #{I18n.t("notification.template_version.new_version_notification")}",
+      "object_data" => {
+        "template_version_id" => id,
+      },
+    }
+  end
+
+  def previous_version
+    requirement_template
+    .template_versions
+    .where(
+      "version_date <=? AND status = ? AND deprecation_reason = ?",
+      template_version.version_date,
+      TemplateVersion.statuses[:deprecated],
+      TemplateVersion.deprecation_reasons[:new_publish],
+    )
+    .where.not(id: template_version.id)
+    .order(version_date: :desc, created_at: :desc)
+    .first
```

Comment:
why not just reuse the following in TemplateVersioningService.

```
 def self.previous_published_version(template_version)
    template_version
      .requirement_template
      .template_versions
      .where(
        "version_date <=? AND status = ? AND deprecation_reason =
 ?",
        template_version.version_date,
        TemplateVersion.statuses[:deprecated],
        TemplateVersion.deprecation_reasons[:new_publish],
      )
      .where.not(id: template_version.id)
      .order(version_date: :desc, created_at: :desc)
      .first
  end
```
<img width="922" alt="Screenshot 2024-05-28 at 11 19 38 AM" src="https://github.com/bcgov/HOUS-permit-portal/assets/32022335/86d3242d-8380-4d4d-a96d-39c9a6c34d7f">

Just make the class method public

Reply:
The point of the service is to abstract away some of this logic into the service. You can call the service from the this model method

Reply:
ok